### ISSUE TITLE
Password hash does not seem to handle special characters

### ISSUE THREAD

**Issue by AdamVig** [usefathom/fathom#95]
I registered my admin user with a secure 25-character password generated by LastPass then attempted to log in through the web interface, but kept getting `401` errors. The password I used contained letters, numbers, and special characters. It also happened to both start and end with special characters.

When I deleted my user and registered again with a password that did not contain special characters, I was able to log in without any problems.

I am willing to take a look at fixing this myself, but figured it is worth asking if the maintainer has any inkling of where this problem may lie.

**Comment by jhabdas** [comment]
Did you try using a [secure password](https://www.xkcd.com/936/) when registering?

**Comment by dewey** [comment]
How is that related to the issue?

**Comment by dannyvankooten** [comment]
Hi @AdamVig,

Sorry if this sounds simplistic, but did you surround the —password argument with double quotes?

**Comment by AdamVig** [comment]
@dannyvankooten I did!

**Comment by dewey** [comment]
I tried to reproduce it but can't get it to fail, I also briefly looked at the code and couldn't find anything suspicious. The password option I used was:

```
--password="*jidiio3890&0293jiodjfnjijsadfdij&"
```

Are you able to reproduce it if you try again with various passwords @AdamVig?

**Comment by jhabdas** [comment]
> How is that related to the issue?

In a secure (read: hard to crack) password special characters become unnecessary. Specials are also exceptionally difficult to enter in some mobile browsing contexts, causing UX to suffer.

Given the likelihood a password with special characters was generated by a password manager and is easy to crack

**Comment by jhabdas** [comment]
> How is that related to the issue?

It's a workaround to help Adam and others move forward while the issue is addressed.

**Comment by jhabdas** [comment]
Adam, have you tried typing the password generated by hand or pasting it from the clipboard using a plain text paste to strip any control characters, e.g. `\n`?

**Comment by jhabdas** [comment]
Adam, since the LastPass Cask [wasn't installing](https://github.com/Homebrew/homebrew-cask/pull/48941) the blessed version of LastPass from the Mac App store but a browser extension instead it seems as though the problem may be related to some recent changes in your environment.

**Comment by dannyvankooten** [comment]
I can't seem to reproduce this either, testing with `fathom register --email="test@dvk.co" --password="[];'\,./<>?:|{}@#$%^&*()_+-=~"` worked just fine for me. So did `--password="\"[];'\,./<>?:|{}@#$%^&*()_+-=~"` (escaped double quote).

One possible gotcha is that your shell might be trying to interpret certain special characters and perform actions based off of it. 

For example, zsh will search for events if you use `!` or start a multi-line string if you use a backtick, even if it's inside double quotes. 

Would it be possible to share the specific password that was failing for you @AdamVig?

**Comment by dannyvankooten** [comment]
Because it's been 9 days and no one has been able to reproduce this issue as of yet, I'm going to close this issue. In case it does persist, feel free to re-open it with details that would help us reproduce. 

Cheers!

**Comment by AdamVig** [comment]
@dannyvankooten Thanks for your patience! I tried this again and it seems that your idea about Zsh being the problem was correct.

When I tried it with a password containing an exclamation point, both with double quotes and without, it spat out the same command back at me with some text inserted where the exclamation point was. Before, the behavior was silent (or I simply did not notice it).

Single quotes fixed the problem for me. Perhaps changing the documentation to always use single quotes would help, but maybe this ticket itself will serve as sufficient documentation.

Thanks for the help and input, everyone!
